standard/lpdb/count.lua

```diff
@@ -0,0 +1,188 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Count
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local Class = require('Module:Class')
+local Logic = require('Module:Logic')
+local String = require('Module:StringUtils')
+local Team = require('Module:Team')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local Count = {}
+
+
+function Count.games(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+
+	return Count._query('game', lpdbConditions)
+end
+
+function Count.matches(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('match', lpdbConditions)
+end
+
+function Count.tournaments(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args, true)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('tournament', lpdbConditions)
+end
+
+function Count.placements(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	if not Logic.readBool(args.includeShowmatch) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Showmatch')}
+	end
+
+	if not Logic.readBool(args.includeQualifier) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatier'), Comparator.neq, 'Qualifier')}
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Qualifier')}
+	end
+
+	if Logic.readBool(args.noEmptyPrizePool) then
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.neq, '')}
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.gt, 0)}
+	end
+
+	if String.isNotEmpty(args.player) then
+		local opponent = mw.ext.TeamLiquidIntegration.resolve_redirect(args.player)
+		local teamConditions = ConditionTree(BooleanOperator.any)
+		for index = 1, 10 do
+			teamConditions:add{
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponent),
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponent:gsub(' ', '_'))
+			}
+		end
+		lpdbConditions:add{teamConditions}
+	end
+
+	if String.isNotEmpty(args.team) then
+		local opponent = Team.queryRaw(args.team).page
```

Comment:
imo use the historical version here
(see e.g. earnings or results table)

Reply:
The results table method queries `opponenttemplate` which doesn't have underscores. Do you mean passing `args.team` and  `args.team:gsub(' ', '_')` to the Team.queryRaw() function?

Reply:
- `Team.queryHistorical(opponentTemplate)` queries the templates (used in results table)
- `Team.queryHistoricalNames(team)` queries pagenames (used in earnings) <-- for those you would have to do the underscore stuff

Reply:
So it there a downside to not using templates as the query parameter? Given that its set for placement data?

Reply:
i personally prefer pagenames

if tt is set up bad it can cause issues for both

if there are some tts missing redirects and instead using duplicates the pagenames still catch it while the query via templatename can not

Reply:
Have adjusted to query opponentname